src/allocation_engine/interval_tree.rs

```diff
@@ -399,6 +399,30 @@ impl InnerNode {
         self.update_cached_height();
         Ok(self.rotate())
     }
+
+    /// Update the state of an old node. This method should be used when we
```

Comment:
Hmmm I don't understand this part. The `insert` method already receives the `node_state` as a parameter. Why would you use this function then?

Reply:
Updated the description to clarify things.